pallets/external-agents/src/lib.rs

```diff
@@ -336,22 +346,36 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 
-    /// Ensure that `agent` is an agent of `ticker` and set the group to `group`.
+    /// Add, remove or modify group of an agent.
     fn try_mutate_agents_group(
         ticker: Ticker,
         agent: IdentityId,
         group: Option<AgentGroup>,
     ) -> DispatchResult {
         GroupOfAgent::try_mutate(ticker, agent, |slot| {
-            ensure!(slot.is_some(), Error::<T>::NotAnAgent);
-
             match (*slot, group) {
```

Comment:
The old structure was simpler. I'd prefer adding another `if group.is_none() { AgentOf::remove(...) }` below the `try_mutate`, keeping the current `match` untouched, and purely handling addition in `unchecked_add_agent`.